osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -191,19 +190,18 @@ protected override void Update()
         {
             base.Update();
 
-            if (fetchedInitialScores && nextPageRequest == null)
+            if (lastFetchCompleted)
             {
+                APIRequest nextPageRequest = null;
+
                 if (ScorePanelList.IsScrolledToStart)
                     nextPageRequest = FetchNextPage(-1, fetchScoresCallback);
                 else if (ScorePanelList.IsScrolledToEnd)
                     nextPageRequest = FetchNextPage(1, fetchScoresCallback);
 
                 if (nextPageRequest != null)
                 {
-                    // Scheduled after children to give the list a chance to update its scroll position and not potentially trigger a second request too early.
```

Comment:
Did you confirm that this reasoning is no longer a requirement?

Reply:
This entire code has changed. Yes, I've confirmed it. There's tests for this as well (`TestScenePlaylistsResultsScreen` and the ones which test scroll to left/scroll to right with delayed load).